vllm/utils.py

```diff
@@ -315,7 +322,13 @@ async def producer(i: int, iterator: AsyncIterator[T]):
     async def consumer():
         try:
             while not all(finished) or not queue.empty():
```

Comment:
I think we can eliminate the `finished` list and just have a count that's decremented in the consumer when it receives each `ProducerFinished`?